.changeset/lemon-phones-sort.md

```diff
@@ -2,8 +2,21 @@
 "@vercel/edge-config": major
 ---
 
-- Use stale-while-revalidate during development
-- Don't throw generic errors
-- Make the return values read only
-- Use the `privateEdgeConfig` global when available
-- Export a clone function to support the read only behaviour
+ - **Return values are now read-only to improve in-memory caching**
+   In addition to the read-only change, the `@vercel/edge-config` package now also exports a `clone` function that can be used to make the read-only values modifiable again. Applications that used to modify the return value must now clone the value before doing the required changes.
+
+   ```typescript
+    import { get, clone } from '@vercel/edge-config';
+
+    const myArray = await get('listOfAllowedIPs');
+    // myArray.push('127.0.0.1'); // This won't work and might throw an error
```

Comment:
```suggestion
    // myArray.push('127.0.0.1'); // This won't work and will throw an error
```

Reply:
It'll cause a TypeScript error a build time, but at runtime it'd only throw in certain cases when the object is truly not writeable beyond just the TypeScript type.

Reply:
Can our proxy object catch this and ensure we throw?

Reply:
@dferber90 it does, the reason why it only "might" throw is because the global that creates the Proxy object is not always present, only in Serverless Functions.